sdk/python/feast/registry.py

```diff
@@ -0,0 +1,123 @@
+        feature_table_proto = feature_table.to_proto()
+        def updater(registry_proto: RegistryProto):
+            for idx, existing_feature_table_proto in enumerate(registry_proto.feature_tables):
+                if existing_feature_table_proto.spec.name == feature_table_proto.spec.name:
+                    registry_proto.feature_tables[idx] = feature_table_proto
+                    return registry_proto
+            registry_proto.feature_tables.append(feature_table_proto)
+            return registry_proto
+        self._update_registry(updater)
+        return
+
+    def list_feature_tables(self):
+        registry_proto = self._get_registry()
+        feature_tables = []
+        for feature_table_proto in registry_proto.feature_tables:
+            feature_tables.append(FeatureTable.from_proto(feature_table_proto))
+        return feature_tables
+
+    def get_feature_table(self, name: str):
+        registry_proto = self._get_registry()
+        for feature_table_proto in registry_proto.feature_tables:
+            if feature_table_proto.spec.name == name:
+                return FeatureTable.from_proto(feature_table_proto)
+        raise Exception(f"Feature table {name} does not exist")
+
+    def delete_feature_table(self, name: str):
+        def updater(registry_proto: RegistryProto):
+            for idx, existing_feature_table_proto in enumerate(registry_proto.feature_tables):
+                if existing_feature_table_proto.spec.name == feature_table_proto.spec.name:
+                    del registry_proto.feature_tables[idx]
+                    return registry_proto
+            raise Exception(f"Feature table {name} does not exist")
+        self._update_registry(updater)
+        return
+
+    @abstractmethod
+    def _get_registry(self):
+        pass
+
+    @abstractmethod
+    def _update_registry(self, updater: Callable[[RegistryProto], RegistryProto]):
+        pass
+
+
+class LocalRegistry(Registry):
+    def __init__(self, filepath: str):
+        self._filepath = Path(filepath)
+        if not self._filepath.exists():
+            registry_proto = RegistryProto()
+            registry_proto.registry_schema_version = REGISTRY_SCHEMA_VERSION
+            registry_proto.version = str(uuid.uuid4())
+            registry_proto.last_updated.FromDatetime(datetime.utcnow())
+            self._filepath.write_bytes(registry_proto.SerializeToString())
+        return
+
+    def _get_registry(self):
+        registry_proto = RegistryProto()
+        registry_proto.ParseFromString(self._filepath.read_bytes())
+        return registry_proto
+
+    def _update_registry(self, updater: Callable[[RegistryProto], RegistryProto]):
+        registry_proto = self._get_registry()
+        registry_proto = updater(registry_proto)
+        registry_proto.version = str(uuid.uuid4())
+        registry_proto.last_updated.FromDatetime(datetime.utcnow())
+        self._filepath.write_bytes(registry_proto.SerializeToString())
```

Comment:
Is it a bit weird to use `_get_registry()` which does reading from a file path, but then inside of the `_update_registry()` we have direct access to the file path? Why do we even need _get_registry then?